storage/include/storage/db_interface.h

```diff
@@ -124,6 +134,18 @@ class IDBClient {
   virtual IDBClientIterator* getIterator() const = 0;
   virtual Status freeIterator(IDBClientIterator* _iter) const = 0;
   IDBClientIterator::Guard getIteratorGuard() const { return IDBClientIterator::Guard{this}; }
+
+ protected:
+  // Storage metrics
+  // TODO: all metrics have announced as mutable because the get methods are const
+  mutable concordMetrics::Component metrics_;
+  mutable concordMetrics::CounterHandle keys_reads_;
+  mutable concordMetrics::CounterHandle total_read_bytes_;
+  mutable concordMetrics::CounterHandle keys_writes_;
+  mutable concordMetrics::CounterHandle total_written_bytes_;
+  mutable int ops_count = 0;
+  const int update_interval = 100;  // TODO: We update the metrics each 100 ops. Currently this is hardcoded value in
+                                    // the future we may add it to the configuration file
```

Comment:
I prefer we don't have any implementation details in the interface. Also, having a public constructor for the interface wouldn't make much sense as users can't properly construct an object via it. Can we maybe separate these details in another class and either contain it or inherit from it in RocksDB and memorydb implementations?

Reply:
done